packages/frontend/src/components/dialogs/ConfigureProgressDialog.tsx

```diff
@@ -71,27 +100,52 @@ export function ConfigureProgressDialog({
     () => {
       ;(async () => {
         try {
-          let isFirstOnboarding = true
-          const configuration: Credentials = {
-            ...credentials,
+          if (!credentials && !qrCode) {
+            throw new Error(
+              'ConfigureProgressDialog needs either credentials or a qrCode'
+            )
           }
+          const configuration: Credentials = credentials || defaultCredentials()
+
           const { proxyEnabled, proxyUrl, ...transportConfig } = configuration
           // Set proxy settings only if neccessary!
+          // but before calling addTransport since it might be needed there (TODO)
           if (proxyUpdated) {
             await BackendRemote.rpc.batchSetConfig(accountId, {
               proxy_enabled:
                 proxyEnabled === true ? Proxy.ENABLED : Proxy.DISABLED,
               proxy_url: proxyUrl,
             })
           }
-          if (
+          if (qrCode) {
+            // create a new transport for accountId based on the QR code
+            await BackendRemote.rpc.addTransportFromQr(accountId, qrCode)
+            // Select 'Device Messages' chat as the initial one. This will serve
+            // as a first introduction to the app after they've entered
+            const deviceChatId = await getDeviceChatId(accountId)
+            if (deviceChatId) {
+              await saveLastChatId(accountId, deviceChatId)
+              // SettingsStoreInstance is reloaded the first time the main screen is shown
+            }
```

Comment:
if I understand the code flow correctly this won't be called for classic email login anymore?

Reply:
Is fixed